### PR TITLE
Add Date-Header to mails

### PR DESCRIPTION
**Pull request by Mirclus**
Added a Date-Header for mails. Noticed it is missing, as a new mail-client showed a week old mail as recently received.

### CODE REVIEW

**Comment by umbrae** [comment]
:eyeglasses: @bsimpson63

File: r2/r2/models/mail_queue.py [hunk 1]
```diff
@@ -428,6 +437,7 @@ def utf8(s, reject_newlines=True):
             msg['To']      = utf8(self.to_addr)
             msg['From']    = utf8(fr)
             msg['Subject'] = utf8(self.subject)
+            msg['Date']    = utf8(self.get_date_string())
```
Comment (bsimpson63) [inline r2/r2/models/mail_queue.py]: It looks like the python standard library has a way to do this: https://docs.python.org/2/library/email.util.html#email.utils.formatdate

You could do something like:
```
timestamp = time.mktime(self.date.utctimetuple())
msg['Date'] = utf8(email.utils.formatdate(timestamp))
```

From some quick googling it's unclear whether we'd want the date to be in the server's timezone, or if just using UTC would be acceptable.

**Comment by bsimpson63** [comment]
:nail_care: (see https://github.com/reddit/reddit/wiki/The-Code-Salon)

**Comment by Mirclus** [comment]
:haircut: I changed the format to the method in email.utils. I used `timetuple` instead of `utctimetuple` as in my test VM the time was off by one hour (DST?).

Just for information, do you want to see the history in the PR or should I squash the commits?

File: r2/r2/models/mail_queue.py [hunk 2]
```diff
@@ -433,6 +435,7 @@ def utf8(s, reject_newlines=True):
             msg['To']      = utf8(self.to_addr)
             msg['From']    = utf8(fr)
             msg['Subject'] = utf8(self.subject)
+            msg['Date'] = utf8(email.utils.formatdate(time.mktime(self.date.timetuple())))
```
Comment (bsimpson63) [inline r2/r2/models/mail_queue.py]: sorry to nitpick, but this line is too long: https://github.com/reddit/styleguide/blob/master/python/pep-0008.rst#maximum-line-length

**Comment by bsimpson63** [comment]
Thanks, looks good other than that one issue.

You can squash the commits together.

**Comment by bsimpson63** [comment]
:nail_care:

**Comment by Mirclus** [comment]
:haircut:

**Comment by bsimpson63** [comment]
:fish: thanks, I'll get this merged.

**Comment by bsimpson63** [comment]
Merged as 1837221fc6909f16bb88b39a991abd3a73e0d0ad.

If you tell me your reddit username I'll give you a trophy.

**Pull request closed without merging**